### PR TITLE
Revise the PR template

### PR DESCRIPTION
**Pull request by kwankyu**
### :books: Description

Made small edits to the PR template, making it concise and tidy overall.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,29 +1,28 @@
-<!-- ^^^^^
-Please provide a concise, informative and self-explanatory title.
-Don't put issue numbers in there, do this in the PR body below.
-For example, instead of "Fixes #1234" use "Introduce new method to calculate 1+1"
--->
-### 📚 Description
+<!-- Please provide a concise, informative and self-explanatory title. -->
+<!-- Don't put issue numbers in the title. Put it in the Description below. -->
```
Comment (tobiasdiez) [inline .github/PULL_REQUEST_TEMPLATE.md:2]: No strong preference, but maybe we should leave the example in here for the time being?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,29 +1,28 @@
...
 
-### ⌛ Dependencies
-<!-- List all open pull requests that this PR logically depends on -->
-<!--
-- #xyz: short description why this is a dependency
-- #abc: ...
+### :hourglass: Dependencies
+
+<!-- List all open PRs that this PR logically depends on
+Depends on #12345: short description why this is a dependency
```
Comment (tobiasdiez) [inline .github/PULL_REQUEST_TEMPLATE.md]: Being a list has the advantage that github automatically expands it, and shows the status (merged or not). See https://github.com/sagemath/sage/pull/35114 for an example. This is not the case of the "Depends on" format. I would leave this for now as a list, until we indeed formalize this using a github action.

**Review by tobiasdiez — CHANGES_REQUESTED** [review]
Thanks!

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (kwankyu) [reply .github/PULL_REQUEST_TEMPLATE.md]: Okay. Done

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (kwankyu) [reply .github/PULL_REQUEST_TEMPLATE.md:2]: Okay. Done.

**Review by tobiasdiez — APPROVED** [review]
LGTM, thanks!

**Comment by kwankyu** [comment]
Thanks!

**Pull request merged**